keps/sig-node/20190604-rootless.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Rootless mode
+authors:
+  - "@AkihiroSuda"
+owning-sig: sig-node
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-06-04
+last-updated: 2020-01-28
+status: provisional
+see-also:
+  - https://github.com/kubernetes/enhancements/pull/1370
+replaces:
+  - https://github.com/kubernetes/enhancements/pull/1084
+---
+
+# Rootless mode
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [FAQ: why not use admission controllers?](#faq-why-not-use-admission-controllers)
+- [Goals](#goals)
+- [Implementation Details](#implementation-details)
+  - [User namespaces](#user-namespaces)
+    - [Paths](#paths)
+    - [Pod SecurityContext](#pod-securitycontext)
+    - [Required changes to Kubernetes](#required-changes-to-kubernetes)
+  - [cgroup](#cgroup)
+    - [Required changes to Kubernetes](#required-changes-to-kubernetes-1)
+  - [Network namespaces](#network-namespaces)
+    - [Required changes to Kubernetes](#required-changes-to-kubernetes-2)
+- [Risks and Mitigations](#risks-and-mitigations)
+- [Graduation Criteria](#graduation-criteria)
+- [Testing](#testing)
+- [History](#history)
+<!-- /toc -->
+
+## Summary
+
+Allow running the entire Kubernetes components (`kubelet`, CRI, OCI, CNI, and all `kube-*`) as a non-root user on the host.
+
+See [FOSDEM 2019 talk "Rootless Kubernetes"](https://www.slideshare.net/AkihiroSuda/rootless-kubernetes) by [@AkihiroSuda](https://github.com/AkihiroSuda) and [@giuseppe](https://github.com/giuseppe) for the overview of Rootless mode.
+See [Usernetes](https://github.com/rootless-containers/usernetes) for the POC.
+
+Rootless mode has been already adopted by k3s, though it doesn't support setting resource limitation as it lacks support for cgroup.
+
+In this KEP, the resource limitation is planned to be supported using cgroup v2, which officially supports delegating its control to a non-root user.
```

Comment:
Should we wait for cgroup v2 being settle down before exploring rootless then?

Reply:
cgroup2 is ready now